pkg/check/util.go

```diff
@@ -178,44 +184,59 @@ func ConvertToPlatformSpec(infra *ocpv1.Infrastructure, checkContext *CheckConte
 				// we need to check if we really need to add to VCenters and FailureDomains
 				vcenter := vCentersToMap(checkContext.PlatformSpec.VCenters)
 
-				// vcenter is missing from the map, add it...
-				if _, ok := vcenter[config.Workspace.VCenterIP]; !ok {
-					convertIntreeToPlatformSpec(config, checkContext.PlatformSpec)
+				// vcenter is missing from the map, add it...  This should not happen.  In multi vCenter we require this
+				// to be configured correctly.  Keeping this loop simple without adding error.
+				for _, vCenter := range checkContext.VCenters {
+					if _, ok := vcenter[vCenter.VCenterName]; !ok {
+						klog.Warningf("vCenter %v is missing from vCenter map", vCenter.VCenterName)
+						convertIntreeToPlatformSpec(config, vCenter.VCenterName, checkContext.PlatformSpec)
+					}
 				}
 
 				if len(checkContext.PlatformSpec.FailureDomains) == 0 {
 					addFailureDomainsToPlatformSpec(config, checkContext.PlatformSpec)
 				}
 			} else {
-				convertIntreeToPlatformSpec(config, checkContext.PlatformSpec)
+				// If we are here, infrastructure resource hasn't been updated for any vCenter.  For multi vcenter support,
+				// being here is not supported.  For 1 vCenter we will allow.
+				if len(checkContext.VCenters) == 1 {
+					for _, vCenter := range checkContext.VCenters {
+						convertIntreeToPlatformSpec(config, vCenter.VCenterName, checkContext.PlatformSpec)
+					}
```

Comment:
simplify with:
convertIntreeToPlatformSpec(config, vCenter[0].VCenterName, checkContext.PlatformSpec) ?

Reply:
checkContext.VCenters is a map.  Like the other review comment, I can either update this to default to the `checkContext.VMConfig.LegacyConfig.Workspace.VCenterIP` to get the old legacy default vcenter definition, or leave w/ loop getting only key.